command/host_create.go

```
@@ -50,7 +43,21 @@ func (c *createHost) ParseParams(params map[string]interface{}) error {
 	return errors.Wrapf(decoder.Decode(params), "error parsing '%s' params", c.Name())
 }
 
+func (c *createHost) parseParamsFromFile(fn string, conf *model.TaskConfig) error {
+	if !filepath.IsAbs(fn) {
+		fn = filepath.Join(conf.WorkDir, fn)
+	}
+	return errors.Wrapf(utility.ReadYAMLFile(fn, &c.CreateHost),
+		"error reading from file '%s'", fn)
+}
+
 func (c *createHost) expandAndValidate(conf *model.TaskConfig) error {
+	// if a filename is defined, then parseParams has not parsed the parameters yet,
+	// since the file was not yet available. it therefore needs to be parsed now.
+	if c.CreateHost.File != "" {
+		c.parseParamsFromFile(c.CreateHost.File, conf)
```

Should return an error if this errors